Lib/gftools/util/google_fonts.py

```diff
@@ -459,3 +462,85 @@ def LicenseFromPath(path):
   """
   return _EntryForEndOfPath(path, _KNOWN_LICENSE_DIRS)
 
+
+def LoadLanguages(languages_dir):
+  languages = {}
+  for textproto_file in glob.iglob(os.path.join(languages_dir, '*.textproto')):
+    with open(textproto_file, 'r', encoding='utf-8') as f:
+      language = text_format.Parse(f.read(), fonts_pb2.LanguageProto())
+      languages[language.id] = language
+  return languages
+
+
+def LoadScripts(scripts_dir):
+  scripts = {}
+  for textproto_file in glob.iglob(os.path.join(scripts_dir, '*.textproto')):
+    with open(textproto_file, 'r', encoding='utf-8') as f:
+      script = text_format.Parse(f.read(), fonts_pb2.ScriptProto())
+      scripts[script.id] = script
+  return scripts
+
+
+def LoadRegions(regions_dir):
+  regions = {}
+  for textproto_file in glob.iglob(os.path.join(regions_dir, '*.textproto')):
+    with open(textproto_file, 'r', encoding='utf-8') as f:
+      region = text_format.Parse(f.read(), fonts_pb2.RegionProto())
+      regions[region.id] = region
+  return regions
+
+
+def SupportedLanguages(
+  ttFont,
+  languages=LoadLanguages(resource_filename("gftools", "lang/languages"))
+  ):
+  """Get languages supported by given ttFont.
+
+  Languages are pulled from the given set. Based on whether exemplar character
+  sets are present in the given font.
+
+  Logic based on Hyperglot: https://github.com/rosettatype/hyperglot/blob/3172061ca05a62c0ff330eb802a17d4fad8b1a4d/lib/hyperglot/language.py#L273-L301
+  """
+  chars = [chr(c) for c in ttFont["cmap"].getBestCmap()]
+  supported = []
+  for lang in languages.values():
+    if not lang.HasField('exemplar_chars') or not lang.exemplar_chars.HasField('base'):
+      continue
+    base = parse.parse_chars(lang.exemplar_chars.base,
+                             decompose=False,
+                             retainDecomposed=False)
+    if set(base).issubset(chars):
+      supported.append(lang)
+  return supported
+
+
+def GetExemplarFont(family):
+  assert len(family.fonts) > 0, 'Unable to select exemplar in family with no fonts: ' + family.name
+  for font in family.fonts:
+    if font.style == 'normal' and font.weight == 400:
+      # Prefer default style (Regular, not Italic)
+      return font
+  return family.fonts[0]
+
+
+def language_comments(languages):
```

Comment:
For consistency, this function should probably be named something closer to LanguageComments

Reply:
Also, can you add a comment explaining what this function is for? Now that the function lives over here, I don't think its purpose is obvious.